src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -132,20 +131,20 @@ def get_counts_for_project(self, project_id: int) -> Iterable[base.BucketedCount
         This may throw an exception if there is some sort of issue fetching counts from the redis
         store.
         """
-        key_prefix = f"{self._counter_key_prefix()}:{project_id}:"
+        bucket_size = self._counter_bucket_size
+        now = int(time.time())
```

Comment:
I think it would be a good idea to grab this from a parameter, e.g. whatever's invoking this should provide the ending timestamp to the range of time they're interested in grabbing counts for. This change should apply to and work for the `_duration` equivalent of this method as well.

Making this change should make testing this method a little easier, as you don't have to manipulate `time.time()` and/or the `time` module to test that this logic works. 

Forcing the caller to provide the timestamp also removes the assumption that the time of execution is actually the time period of interest; If there's some sort of significant delay in between when this is executed and when the parent is invoked, this makes it easier to handle that situation. You could say that this is a form of dependency inversion.

Reply:
I  did it with `timestamp` as a parameter now. The code is just so much cleaner that way.